### PR TITLE
ASSETS-88900 : Update Pagination for gmo-campiagn-list Block

### PR DESCRIPTION
**Pull request by TyroneAEM**
Workfront Task
https://experience.adobe.com/#/@wfadoberm/so:adoberm-Production/workfront/task/6619b7fb00386db714fe67602e7ece4f/overview

Test URLs:

https://assets-88900--adobe-gmo--hlxsites.hlx.page/drafts/tyrone/campaigns#

Screenshot 2024-04-23 at 2 33 53 PM

### CODE REVIEW

File: blocks/gmo-campaign-list/gmo-campaign-list.js [hunk 1]
```diff
@@ -153,6 +194,10 @@ function buildCampaignList(campaigns, numPerPage) {
     listWrapper.dataset.totalresults = campaigns.length;
 
     campaigns.forEach((campaign, index) => {
+
+    //Todo delete console statement
+    console.log('index ',index);
```
Comment (shiv-gup) [inline blocks/gmo-campaign-list/gmo-campaign-list.js:199]: @TyroneAEM Is it still needed?

File: blocks/gmo-campaign-list/gmo-campaign-list.js [hunk 2]
```diff
@@ -330,126 +384,68 @@ function buildListFooter(rows, rowsPerPage) {
...
 }
 
-function changePage(targetPage) {
-    // if targetPage = currentPage, do nothing
-    const nextBtn = document.querySelector('.footer-pagination-button.next');
-    const prevBtn = document.querySelector('.footer-pagination-button.prev');
-    const currentPageBtn = document.querySelector('#current-page');
-    const currentPage = parseInt(currentPageBtn.dataset.pagenumber);
+function nextPage(nextBtn) {
+console.log('currentPageInfo');
```
Comment (shiv-gup) [inline blocks/gmo-campaign-list/gmo-campaign-list.js:408]: can you remove console logs

File: blocks/gmo-campaign-list/gmo-campaign-list.js [hunk 3]
```diff
@@ -330,126 +384,68 @@ function buildListFooter(rows, rowsPerPage) {
...
-    const currentPageValue = parseInt(currentPageBtn.dataset.pagenumber);
-    const targetPage = (currentPageValue + 1);
-    changePage(targetPage);
-    prevBtn.classList.add('active');
 }
 
 function prevPage(prevBtn) {
-    if (!(prevBtn.classList.contains('active'))) {
-        return;
+console.log('currentPageInfo');
```
Comment (shiv-gup) [inline blocks/gmo-campaign-list/gmo-campaign-list.js:429]: remove console logs

File: blocks/gmo-campaign-list/gmo-campaign-list.js (hunk 2 above)
Comment (TyroneAEM) [reply blocks/gmo-campaign-list/gmo-campaign-list.js]: Removed

File: blocks/gmo-campaign-list/gmo-campaign-list.js (hunk 3 above)
Comment (TyroneAEM) [reply blocks/gmo-campaign-list/gmo-campaign-list.js]: Removed

File: blocks/gmo-campaign-list/gmo-campaign-list.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { readBlockConfig } from '../../scripts/lib-franklin.js';
 import { decorateIcons } from '../../scripts/lib-franklin.js';
-import { graphqlAllCampaigns } from '../../scripts/graphql.js';
+import { graphqlAllCampaigns, graphqlCampaignCount } from '../../scripts/graphql.js';
 
 const icon = 'https://delivery-p108396-e1046543.adobeaemcloud.com/adobe/assets/deliver/urn:aaid:aem:acdaa42f-00ae-42f4-97e5-8309c42d9076/marketing-hub-102023-lockup-video.png'
 const testCampaigns = [
```
Comment (shiv-gup) [inline blocks/gmo-campaign-list/gmo-campaign-list.js:6]: Can you remove this `testCampaigns`

File: blocks/gmo-campaign-list/gmo-campaign-list.js [hunk 5]
```diff
@@ -123,14 +123,33 @@ const testConfig = [
...
+    currentPageInfo.itemCount = campaigns.length;
 
-export default async function decorate(block) {
-    const campaignResponse = await graphqlAllCampaigns();
-    console.log(campaignResponse);
-    const campaigns = campaignResponse.data.campaignList.items;
-    console.log(campaignResponse.data.campaignList.items);
-    const numPerPage = 4; 
-    const campaignCount = campaigns.length;
     const config = testConfig;
```
Comment (shiv-gup) [inline blocks/gmo-campaign-list/gmo-campaign-list.js:153]: remove `testConfig`

**Review by shiv-gup — APPROVED** [review]

File: blocks/gmo-campaign-list/gmo-campaign-list.js (hunk 5 above)
Comment (TyroneAEM) [reply blocks/gmo-campaign-list/gmo-campaign-list.js:153]: I think testConfig is used to provide test/dummy data at the moment for the listHeaders, and will be replaced by data from graphql in future Workfront Task

**Pull request merged**